### PR TITLE
chore: remove storybook

### PR DESCRIPTION
**Pull request by hofmann-felix**
### What's included?
* Since Storybook has not been updated for a long time and was hardly used, we decided to remove it. For this the following files were deleted or changed:
  * Delete all stories (`.mdx` and `.tsx`)
  * Delete configuration in `.storybook`
  * Change `build.yml`
  * Remove all Storybook related dependencies

### CODE REVIEW

File: README.md
Comment (hofmann-felix) [inline README.md:10]: The project needs >= 18 for quite some time now. I took the opportunity to adjust the readme.

**Review by georgiee — APPROVED** [review]
Nothing to add. Thanks for removing it 👍
CC @feedm3 so you're aware of us removing it. Maybe you approve in addition if you agree?

Motivation:
I encouraged Felix to move ahead but it would be good to have a second approval on this. We decided to remove storybook because there are only few folks left who maintain the website and nobody cares for the stroybooks anymore. It was good in the beginning but right now it's useless noise.

**Comment by georgiee** [comment]
Removed the required check for storybook:

Screenshot 2023-03-17 at 13 07 25

**Review by feedm3 — APPROVED** [review]
It's fine, outdated code just slows down the overall development speed. Maintaining story-book also is quite a lot of work, which was worth at the beginning but not right now.

We can always come back to this PR/commit to check the implementation.

**Pull request merged**
